webapp/src/client/app/modules/web-client/vnc/web-client-vnc.component.ts

```diff
@@ -400,4 +398,32 @@ export class WebClientVncComponent extends WebClientBaseComponent implements  On
         return 'Connection error: Please verify your connection settings.';
     }
   }
+
+  private getErrorKind(errorData: UserIronRdpError | string): UserIronRdpErrorKind {
+    if (typeof errorData === 'string') {
+      return this.parseErrorString(errorData);
+    }
+
+    if (errorData?.kind() !== UserIronRdpErrorKind.General) {
+      return errorData.kind();
+    }
+
+    console.error('Error Kind:', errorData.kind());
+    const backtrace: string = errorData?.backtrace();
+    return this.parseErrorString(backtrace);
+  }
+
+  private parseErrorString(error: string): UserIronRdpErrorKind {
+    console.error(error);
+
+    if (error.indexOf('authentication rejected') >= 0) {
+      return UserIronRdpErrorKind.WrongPassword;
+    }
+
+    if (error.indexOf('connection has been rejected') >= 0) {
+      return UserIronRdpErrorKind.LogonFailure;
+    }
+
+    return UserIronRdpErrorKind.General;
+  }
```

Comment:
question: I understand the intention is to parse the backtrace of the error returned by the VNC module. Is that correct? I don’t think this should be a concern for the Devolutions Gateway web UI. The WASM module should return the correct error kind and the consumer code just roll with it. Parsing the error string is very fragile, but I think that in this case we don’t have a choice since that’s the only information returned by the server (no standardized status codes). Nevertheless, the string parsing logic should live in the WASM module.

cc @thenextman , how are we proceeding in FreeVNC? Do you have a list of common strings sent by the major VNC servers that we could reuse?

Reply:
I absolutely understand it is not ideal. The reasoning in parsing the backtrace is a last attempt to give the user any type of an informative error message. 

At the moment, most errors from the components are returning error kind: General. We need the error kind to match up with what the backtrace is giving the webapp. Those parsing phrases above are real case examples of when the error kind is returned as General.

Reply:
@CBenoit No we don't even try. Actually the error handling in FreeVNC is super basic; it's somewhat like the above: allow the consumer to differentiate between a connection failure and an authentication failure. For any more detailed diagnostic in RDM it's necessary to read logs. 

On non-Windows RDM, even RDP error handling is basically non-existent ("an error occurred, read your log file!"). It's a pain point but the related teams don't really see fixing that as a priority 😞

Reply:
@kristahouse I’ll include the parsing in the WASM module directly so you don’t have to do it here.

Reply:
@CBenoit Perfect. Thank you!